clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1423,34 +1422,40 @@ public void resume(Collection<TopicPartition> partitions) {
                 throw new IllegalArgumentException("The target time for partition " + entry.getKey() + " is " +
```

Comment:
Following up on one of Ismael's previous comments. It might be worth a comment here explaining that this explicitly excludes the magic earliest and latest timestamp values. We depend on this in order to ensure that the returned timestamp in `OffsetAndTimestamp` is always positive. Also, do you think it's worth validating the requirement that the timestamp is positive in the `OffsetAndTimestamp` constructor?